File: GXpert/GXpert.Web/Modules/QuestionBank/QuestionCourse/QuestionCourseColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.QuestionBank.Columns;

[ColumnsScript("QuestionBank.QuestionCourse")]
[BasedOnRow(typeof(QuestionCourseRow), CheckNames = true)]
public class QuestionCourseColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string QuestionText { get; set; }
    public string CourseTitle { get; set; }
    public string ClassTitle { get; set; }
    public string SemesterTitle { get; set; }
    public string SubjectTitle { get; set; }
    public string TopicTitle { get; set; }
}
 
   

File: GXpert/GXpert.Web/Modules/ServerTypes/QuestionBank/QuestionCourseColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { QuestionCourseRow } from "./QuestionCourseRow";

export interface QuestionCourseColumns {
    Id: Column<QuestionCourseRow>;
    QuestionText: Column<QuestionCourseRow>;
    CourseTitle: Column<QuestionCourseRow>;
    ClassTitle: Column<QuestionCourseRow>;
    SemesterTitle: Column<QuestionCourseRow>;
}

export class QuestionCourseColumns extends ColumnsBase<QuestionCourseRow> {
    static readonly columnsKey = 'QuestionBank.QuestionCourse';
    static readonly Fields = fieldsProxy<QuestionCourseColumns>();
}